src/lightning_app/components/serve/auto_scaler.py

```diff
@@ -429,7 +429,11 @@ def update_servers(self, server_works: List[LightningWork]):
         AutoScaler uses this method to increase/decrease the number of works.
         """
         old_servers = set(self._servers)
-        server_urls: List[str] = [server.url for server in server_works if server.url]
+        server_urls: List[str] = [
+            f"http://{server._internal_ip}:{server._port}"
```

Comment:
This needs to be the server.port, otherwise it won't be defined.

```suggestion
            f"http://{server._internal_ip}:{server.port}"
```